src/from_substrait.cpp

```diff
@@ -636,7 +638,33 @@ shared_ptr<Relation> SubstraitToDuckDB::TransformReadOp(const substrait::Rel &so
 		} else {
 			scan = GetValuesExpression(sget.virtual_table().expressions());
 		}
-
+	} else if (sget.has_iceberg_table()) {
+		if (sget.iceberg_table().direct().metadata_uri().empty()) {
+			throw InvalidInputException("Metadata file missing in iceberg table read in substrait");
+		}
+		string name = "iceberg_" + StringUtil::GenerateRandomName();
+		named_parameter_map_t named_parameters({});
+		vector<Value> parameters {sget.iceberg_table().direct().metadata_uri()};
+		if (sget.iceberg_table().direct().has_snapshot_id()) {
+			auto str = sget.iceberg_table().direct().snapshot_id();
+			int64_t snapshot_id = strtoimax(str.c_str(), nullptr, 10);
+			if (snapshot_id == 0) {
+				throw InvalidInputException("Invalid snapshot id: " + sget.iceberg_table().direct().snapshot_id());
+			}
+			parameters.push_back(Value::UBIGINT(snapshot_id));
+		} else if (sget.iceberg_table().direct().has_snapshot_timestamp()) {
+			parameters.push_back( Value::TIMESTAMP(timestamp_t(sget.iceberg_table().direct().snapshot_timestamp())));
+		}
+		shared_ptr<TableFunctionRelation> scan_rel;
+		if (acquire_lock) {
```

Comment:
assuming context and context_wrapper are same type, this would probably cleaner with something like:

auto ctx = acquire_lock ? context : context_wrapper;

that way the main body of operations is not repeated.

Reply:
One is a shared_ptr<ClientContext> and the other is a shared_ptr<RelationContextWrapper>.

Reply:
that seems like you're saying they are the same thing

Reply:
Formatting issue, the angle brackets disappeared.

Reply:
Got it, thanks. I'm Resolving.